packages/components/src/combobox-control/README.md

```diff
@@ -92,35 +85,35 @@ If this property is added, a help text will be generated using help property as
 
 The options that can be chosen from.
 
--   Type: `Array<{ value: String, label: String }>`
+-   Type: `Array<{ value: string, label: string }>`
 -   Required: Yes
 
 #### onFilterValueChange
 
-Function called with the control's search input value changes. The argument contains the next input value.
+Function called when the control's search input value changes. The argument contains the next input value.
 
--   Type: `Function`
+-   Type: `( value: string ) => void`
 -   Required: No
 
 #### onChange
 
 Function called with the selected value changes.
 
--   Type: `Function`
+-   Type: `( value: string | null ) => void`
```

Comment:
I have a feeling we won't be able to actually standardize the APIs for our existing components (because back compat). The only thing we can do is agree on a standard moving forward for new components.

> • we should make sure that the value passed as argument of `onChange` has the same type as the `value` prop of the component
> • we should try to also standardize the shape of the callback's first argument

Yes 💯

> • we should standardise the behaviour when the component's value is changed to an empty state (e.g cleared)

This will likely not be a concern for new components, since we're moving away from built in reset buttons. If it does arise though, I still think calling `onChange( undefined )` is the most straightforward and "standardizable" way. This is an overview of the current built-in reset behavior by the way:

### `undefined`

- ColorPalette
- DuotonePicker
- FontSizePicker
- GradientPicker
- RangeControl
- BoxControl (`Record< string, undefined >`)

### `null`

- ComboboxControl
- DateTimePicker

### empty string

- SearchControl

### empty array

- FormTokenField

Reply:
> I have a feeling we won't be able to actually standardize the APIs for our existing components (because back compat). The only thing we can do is agree on a standard moving forward for new components.

Yeah, I have the same suspicion. But as you say, we can at least enforce a standard for future work

> This is an overview of the current built-in reset behavior by the way:

This recap is already so valuable, thank you!